### PR TITLE
Initial implementation: SetDisplayName -> SetMxIdDialog

### PR DESCRIPTION
**Pull request by lukebarnard1**
- Replaces SetDisplayNameDialog with SetMxIdDialog. This new dialog will use InteractiveAuth to authenticate a user with their chosen mxid.

De-scoped:
 - style tweaks for the InteractiveAuth in the dialog (capcha) and error message.
 - checking for mxid availability
2017-04-28-132742_783x567_scrot

For now, based on https://github.com/vector-im/riot-web/issues/3604#issuecomment-293338525

### CODE REVIEW

File: src/components/views/dialogs/SetMxIdDialog.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2016 OpenMarket Ltd
```
Comment (dbkr) [inline src/components/views/dialogs/SetMxIdDialog.js]: copyright

File: src/components/views/dialogs/SetMxIdDialog.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    },
+
+    onSubmit: function(ev) {
+        this.setState({
+            doingUIAuth: true,
+        });
+    },
+
+    _generateAndCachePassword: function() {
+        const pass = Math.random().toString(36).slice(2);
```
Comment (dbkr) [inline src/components/views/dialogs/SetMxIdDialog.js]: hmm, I guess so!

File: src/components/views/dialogs/SetMxIdDialog.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        return pass;
+    },
+
+    _makeRegisterRequest: function(auth) {
+        // Not upgrading - changing mxids
+        const guestAccessToken = null;
+
+        return this._matrixClient.register(
+            this.state.username,
+            this._generateAndCachePassword(),
```
Comment (dbkr) [inline src/components/views/dialogs/SetMxIdDialog.js]: So here you're setting the password before the account is registered (and overwriting it with a new one for each time we try to register). It feels like it really ought to be set along with the access token.

**Comment by dbkr** [comment]
Oh also you seem to have met our new friend, `undefined is not a function`.

File: src/components/views/dialogs/SetMxIdDialog.js (hunk 3 above)
Comment (lukebarnard1) [inline src/components/views/dialogs/SetMxIdDialog.js]: Yep, agreed

**Comment by lukebarnard1** [comment]
It looks like `isGuest` was not being mocked. I'll make it return false. Seems odd that this has never been run into, indicates that only now are the tested code paths calling `isGuest`.

**Pull request merged**